run-functests-on-centos7-vm.sh

```diff
@@ -138,33 +140,49 @@ EOF
     ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-state.sh /dev/vdb Available"
     sudo virsh detach-device "${VM_NAME}" vdb.xml --live
     sleep 1
-    ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-not-exist.sh /dev/vdb"
+    ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-removed.sh /dev/vdb Available"
+
 
     # attach/detach drive in Available and Ready state
     sudo virsh attach-device "${VM_NAME}" vdb.xml --live
     sleep 1
     ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-state.sh /dev/vdb Available"
-    ssh_cmd "${VM_IPADDR}" "./kubectl-direct_csi drive format --drives /dev/vdb"
+    ssh_cmd "${VM_IPADDR}" "./kubectl-directpv drive format --drives /dev/vdb"
     sleep 1
     ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-state.sh /dev/vdb Ready"
     sudo virsh detach-device "${VM_NAME}" vdb.xml --live
     sleep 1
-    ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-not-exist.sh /dev/vdb"
+    ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-removed.sh /dev/vdb Ready"
 
-    # attach/detach drive in Available, Ready, InUse and Terminating state
-    sudo virsh attach-device "${VM_NAME}" vdb.xml --live
+   # attach a new device to test InUse
+    qemu-img create -f raw "${VM_NAME}-vdc.img" 512M
+    cat > vdc.xml <<EOF
+<disk type='file' device='disk'>
+  <source file='${PWD}/${VM_NAME}-vdc.img'/>
+  <target dev='vdc'/>
+  <serial>thisisdeviceC</serial>
+</disk>
+EOF
+
+    # attach/detach drive in Available, Ready, InUse
+    sudo virsh attach-device "${VM_NAME}" vdc.xml --live
     sleep 1
-    ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-state.sh /dev/vdb Available"
-    ssh_cmd "${VM_IPADDR}" "./kubectl-direct_csi drive format --drives /dev/vdb"
+    ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-state.sh /dev/vdc Available"
+    ssh_cmd "${VM_IPADDR}" "./kubectl-directpv drive format --drives /dev/vdc"
     sleep 1
-    ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-state.sh /dev/vdb Ready"
+    ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-state.sh /dev/vdc Ready"
     ssh_cmd "${VM_IPADDR}" "functests/run-deploy-minio.sh"
-    ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-state.sh /dev/vdb InUse"
-    sudo virsh detach-device "${VM_NAME}" vdb.xml --live
+    ssh_cmd "${VM_IPADDR}" "functests/run-check-drive-state.sh /dev/vdc InUse"
+    ssh_cmd "${VM_IPADDR}" "functests/run-check-volume-exist.sh /dev/vdc true"
+
+    sudo virsh detach-device "${VM_NAME}" vdc.xml --live
```

Comment:
Like discussed, we need to wait until the MinIO pods are up. only after that, we need to detach the in-use drive, otherwise the volumes check will not work. 

refer deploy_minio function

Reply:
`functests/run-deploy-minio.sh` actually calls `deploy-minio.sh` and waits until all minio pods are up . so its taken care